.github/workflows/build-image.yml

```diff
@@ -230,22 +290,31 @@ jobs:
         id: manifest
         working-directory: /tmp/digests
         run: |
-          # create manifest list from platform-specific digests
+          # build the list of source images (by digest)
+          sources=""
+          for digest in *; do
+            sources="$sources ${{ env.REGISTRY }}/${{ github.repository }}-cuda-dev@sha256:$digest"
+          done
+
+          # create manifest and tag it
           docker buildx imagetools create \
-            $(echo "${{ steps.meta.outputs.tags }}" | tr '\n' ' ' | sed 's/ */ /g' | \
-              awk '{for(i=1;i<=NF;i++) printf("-t %s ", $i)}') \
-            $(printf '${{ env.REGISTRY }}/${{ github.repository }}-cuda-dev@sha256:%s ' *)
+            -t ${{ env.REGISTRY }}/${{ github.repository }}-cuda-dev${{ matrix.image_suffix }}:pr-${{
+              github.event.pull_request.number }} \
+            -t ${{ env.REGISTRY }}/${{ github.repository }}-cuda-dev${{ matrix.image_suffix }}:sha-${GITHUB_SHA::7} \
```

Comment:
This is tricky - I think the logic we use for the shas are: `$(git rev-parse --short HEAD)` which will produce the smallest unique sha. Sometimes it could be 5 characters, sometimes 12, etc. I am murky on the details but im not certain this will be reliable

Reply:
ack, let's not truncate this then